EXT_BeamPhysics.md

```diff
@@ -0,0 +1,269 @@
+    **branch-index>>element-index** where **branch-index** is the associated
+branch index integer, and **element-index** is the associated lattice element index within the branch.
+
+- `locationInElement`
+  - Type Optional *(int)*
+  - Description: This attribute is used with  `latticeElementID`/`latticeElementName` to specify
+  the origin where the particle or particles are measured with respect in the **lattice** coordinate system.   
+  - The origin is always on the longitudinal axis (x = y = 0) of the **lattice** coordinate system.
+  - Possible values:    
+    - `Upstream-End`: Upstream end of element outside of any edge fields.
+    - `<s-position>`: Where `<s-position>` is a number. Inside the element at a distance, given by `<s-position>`,
+    from the upsteam end of the element.
+    - `Downstream-End`: Downstream end of the element outside of any edge fields.
+  - Note: Since some programs will model edge fields of a lattice element as having zero length, the longitudinal **s**-position
+does not necessarily provide enough information to determine where a particle is with respect to an edge field so `Upstream-End` and `Downstream-End` are provided.
+
+- `momentumNormalization`
+    - Type: Optional *(string)*
+    - Description: Normalization used for momentum values.
+    - Possible values:
+        - `referenceTotalMomentum`: Normalize with respect to the `referenceTotalMomentum` attribute
+        - `referenceEnergy`: Normalize with respect to the `referenceEenergy` attribute.
+        - `none`: No normalization.
+
+- `referenceEnergy`
+    - Type: Optional *(float)* attribute.
+    - Description: Specifies the reference energy from which the `energy` is measured with respect to.
+
+- `referenceTotalMomentum`
+    - Type: Optional *(float)* attribute
+    - Description: Specifies the reference total momentum from which the total momentum is measured with respect to.
+
+- `totalCharge`
+    - Type: Optional *(float)* attribute.
+    - Description: The total charge of all the particles.
+
+Per-Particle Records of the `Particle Root Group`
+---------------------
+
+The following records store data on a particle by particle basis.
+
+- `time-refTime`
+- Type: Optional *(float)*
+- Description: Particle time minus the referece time.
+
+- `energy/`
+  - Type: Optional *(float)*
+  - Description: The total energy of the particles relative to the `referenceEnergy` attribute.
+
+- `electricField/`
+    - Type: Optional 2-component *(float)*
+    - Description: Electric field. Used for photons only.
+    - Components: (`x`, `y`).
+        - For each component, the field is specified using either (`amplitude`, `phase`) or (`Real`, `Imaginary`)
+        subcomponents.
+
+- `macroCharge/`
+    - Type: Optional *(float)*
+    - Description: "Macro"-particles are simulation particles that represent multiple real particles.
+    The `macroCharge` is the the total charge of the collection of particles that a macro-particle
+    represents.
```

Comment:
This seems similar to `weighting` (as well as the attribute `weightingPower`) in the [ED-PIC extension](https://github.com/openPMD/openPMD-standard/blob/upcoming-2.0.0/EXT_ED-PIC.md). 

Should we maybe use the same conventions (and maybe isolate this in an extension specifically for macroparticles)?

Reply:
This is indeed similar to `weighting`. There is also in ED-PIC `weightingPower` which gets multiplied with `weighting` and there is no equivalent to `weightingPower` for Beam Physics. All-in-all I would say that trying to combine these in a third extension is more problematical than it is worth. However, renaming `macroCharge` to `weighting` and making the the definition here consistent with ED-PIC would be OK with me.

Reply:
OK, your suggested solution sounds reasonable to me.

Reply:
Changed to `weighting` in the new draft to be uploaded soon...